### ISSUE TITLE
Stop "fail to restore" occurances with CPS/NuGet conflicts

### ISSUE THREAD

**Issue by rrelyea** [NuGet/Home#3877]
Short of the CPS project lock fix covered by #3807 

This is representing a fix we tried last night and had very good results with testing.
We are just reversing the order of these 2 actions:

**Current order**
1) save project.assets.json
2) save .g.targets and .g.props

**New order**
1) save .g.targets and .g.props
2) save project.assets.json

**Comment by rrelyea** [comment]
From: Sri
Sent: Thursday, November 3, 2016 11:18 PM
Subject: RE: Auto Restore broken -- fix to test is coming
I’ve also verified that the race seems to be gone with these privates. Additionally, the blocking bug below that unit test discovery is broken appears to be because of this issue. In a newly created project I was able to repro the issue but not if I opened and closed the project. With these privates that issue seems to be resolved as well.

287789 Tests are not discovered for .net core test projects 

From: Nat
I’ve been trying out a build with this commitReverse NuGet fix and https://github.com/dotnet/roslyn-project-system/pull/698
So far the broken scenarios in the following appear to be working properly: (roslyn project system bugs)
• #672 Nominate project api is not called for multi-target frameworks
• #682 .NET Core Console application fails to run
• #692 P2P refs aren't working for netcore projects
